docbook/pom.xml

```diff
@@ -106,10 +111,55 @@
                     </options>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>exec-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>process-spring-version</id>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <executable>${project.basedir}/scripts/set_spring_dep_ver.sh</executable>
+                    <environmentVariables>
+                        <BASEDIR>${project.basedir}</BASEDIR>
+                    </environmentVariables>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>dev.jbang</groupId>
+                <artifactId>jbang-maven-plugin</artifactId>
+                <version>0.0.4</version>
+                <executions>
+                    <execution>
+                        <id>run</id>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <jbangVersion>${jbang.ver}</jbangVersion>
+                            <script>scripts/before.java</script>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>cleanup</id>
+                        <phase>install</phase>

```

Comment:
I think this should run at the `compile` phase. That's when the docbook generate goal is invoked.

Reply:
done!